app/models/article.rb

```diff
@@ -416,6 +416,17 @@ def access_by?(user)
     user.admin? || user_id == user.id
   end
 
+  def merge_with(other_article_id)
+    Article.find(other_article_id).comments.each do |comment|
+      comment.article_id = self.id
+      comment.save
+    end
+    merge_article = Article.find(other_article_id)
+    self.body += " " + merge_article.body
+    self.save
```

Comment:
Saving the article and destroying the old one could both have different types of errors. Consider using a transaction to ensure that their both or none happen.